### PR TITLE
Add User and Role classes

### PR DESCRIPTION
**Pull request by Eidenn005**
Add User and Role classes
Add liquibase scripts for DB

### CODE REVIEW

File: src/main/resources/db/changelog/changes/create-users-roles-table.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    foreignKeyName: fk_users_roles_user
+                    references: users(id)
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    foreignKeyName: fk_users_roles_role
```
Comment (oksana-miazina) [inline src/main/resources/db/changelog/changes/create-users-roles-table.yaml:20]: I also added constraint `primaryKey: true` for both, as these pairs shouldn't duplicate

File: src/main/java/model/User.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    @Column(nullable = false)
+    private String password;
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    private Set<Role> roles = new HashSet<>();
+
+    private boolean isDeleted;
```
Comment (oksana-miazina) [inline src/main/java/model/User.java:58]: It seems to me, it should be initialized with `false`, and have `nullable = false`

File: src/main/java/model/User.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+@Entity
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@NoArgsConstructor
+@AllArgsConstructor
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
```
Comment (oksana-miazina) [inline src/main/java/model/User.java]: I suggest omit writing `value` where it is possible
```suggestion
@SQLRestriction("is_deleted=false")
```

**Review by oksana-miazina — APPROVED** [review]

File: src/main/java/model/Role.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (ReamFOX) [inline src/main/java/model/Role.java:20]: I think it is redudant but up to you

File: src/main/java/model/User.java (hunk 2 above)
Comment (ReamFOX) [inline src/main/java/model/User.java:58]: U can set here def value as well.

**Review by ReamFOX — APPROVED** [review]

File: src/main/java/model/User.java (hunk 2 above)
Comment (Eidenn005) [reply src/main/java/model/User.java:58]: done

File: src/main/resources/db/changelog/changes/create-users-roles-table.yaml (hunk 1 above)
Comment (Eidenn005) [reply src/main/resources/db/changelog/changes/create-users-roles-table.yaml:20]: done

File: src/main/java/model/Role.java (hunk 4 above)
Comment (Eidenn005) [reply src/main/java/model/Role.java:20]: Leave as is.

File: src/main/java/model/User.java (hunk 2 above)
Comment (Eidenn005) [reply src/main/java/model/User.java:58]: done

**Pull request merged**